### PR TITLE
doc: added environment vars reference

### PR DESCRIPTION
**Pull request by mshabunin**
Collected most environment variables used to a single document.

### CODE REVIEW

File: doc/tutorials/introduction/env_reference/env_reference.markdown [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+| OPENCV_CORE_PARALLEL_PLUGIN_${NAME} | string, glob | | parallel_for plugin library name (glob), e.g. default for TBB is "opencv_core_parallel_tbb*.so" |
+| OPENCV_DNN_PLUGIN_PATH | paths | | directories to search for _dnn_ plugins |
+| OPENCV_DNN_PLUGIN_${NAME} | string, glob | | parallel_for plugin library name (glob), e.g. default for TBB is "opencv_core_parallel_tbb*.so" |
+| OPENCV_CORE_PLUGIN_PATH - for highgui plugins (YES it is CORE) |
+| OPENCV_UI_PLUGIN_${NAME} |
+| OPENCV_VIDEOIO_PLUGIN_PATH |
+| OPENCV_VIDEOIO_PLUGIN_${NAME} |
+
+### OpenCL
+| name | type | default | description |
```
Comment (asmorkalov) [inline doc/tutorials/introduction/env_reference/env_reference.markdown:145]: Table formatting is broken: https://pullrequest.opencv.org/buildbot/export/pr/23968/docs/d6/dea/tutorial_env_reference.html

File: doc/tutorials/introduction/env_reference/env_reference.markdown [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+| OPENCV_OPENCL_SVM_FORCE_UMAT_USAGE | bool | false | |
+| OPENCV_OPENCL_SVM_DISABLE_UMAT_USAGE | bool | false | |
+| OPENCV_OPENCL_SVM_CAPABILITIES_MASK | num | | |
+| OPENCV_OPENCL_SVM_BUFFERPOOL_LIMIT | num | | same as OPENCV_OPENCL_BUFFERPOOL_LIMIT, but for SVM buffers |
+
+##### Links:
+- https://github.com/opencv/opencv/wiki/OpenCL-optimizations
+
+
+### Tracing/Profiling
```
Comment (asmorkalov) [inline doc/tutorials/introduction/env_reference/env_reference.markdown:183]: Trable formatting is broken. See https://pullrequest.opencv.org/buildbot/export/pr/23968/docs/d6/dea/tutorial_env_reference.html

**Comment by mshabunin** [comment]
I've fixed all tables and :star: symbols. I'm going to re-read and fix the text once more tomorrow.

**Review by asmorkalov — APPROVED** [review]
:+1: LGTM!

**Pull request merged**